tools/PI/DevHome.PI/Helpers/ExternalTool.cs

```diff
@@ -75,50 +78,39 @@ public string OtherArgs
         ArgPrefix = argprefix;
         OtherArgs = otherArgs;
         IsPinned = isPinned;
+        PinGlyph = IsPinned ? CommonHelper.UnpinGlyph : CommonHelper.PinGlyph;
 
         ID = Guid.NewGuid().ToString();
 
         if (!string.IsNullOrEmpty(executable))
         {
-            GetToolImage();
-            GetMenuIcon();
+            GetIcons();
         }
     }
 
-    private async void GetToolImage()
+    partial void OnIsPinnedChanged(bool oldValue, bool newValue)
     {
-        try
-        {
-            _softwareBitmap ??= GetSoftwareBitmapFromExecutable(Executable);
-            if (_softwareBitmap is not null)
-            {
-                ToolIcon = await GetSoftwareBitmapSourceFromSoftwareBitmap(_softwareBitmap);
-            }
-        }
-        catch (Exception ex)
-        {
-            _log.Error(ex, "Failed to get tool image");
-        }
+        PinGlyph = newValue ? CommonHelper.UnpinGlyph : CommonHelper.PinGlyph;
     }
 
-    private async void GetMenuIcon()
+    private async void GetIcons()
     {
         try
         {
             _softwareBitmap ??= GetSoftwareBitmapFromExecutable(Executable);
             if (_softwareBitmap is not null)
             {
-                var bitmapUri = await SaveSoftwareBitmapToTempFile(_softwareBitmap);
-                MenuIcon = new BitmapIcon
-                {
-                    UriSource = bitmapUri,
-                    ShowAsMonochrome = false,
-                };
+                ToolIcon = await GetSoftwareBitmapSourceFromSoftwareBitmap(_softwareBitmap);

```

Comment:
add `async` to the end of the method name.

Reply:
Ok.